websites/K/KOLEO/presence.ts

```diff
@@ -0,0 +1,980 @@
+const presence = new Presence({
+		clientId: "1265368122689458378",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/idEEfM8.png",
+	Logo2 = "https://i.imgur.com/mYXLmgF.png",
+	Buy = "https://i.imgur.com/Z8xQXY4.png",
+	Train = "https://i.imgur.com/Jz7uUX1.png",
+	Ticket = "https://i.imgur.com/PDNRKZ0.png",
+}
+
+let oldStations: string | string[];
+
+function NoPage(presenceData: PresenceData): void {
+	presenceData.name = "KOLEO - 404";
+	presenceData.details = "Nie znaleziono strony.";
+	presenceData.largeImageKey = Assets.Logo;
+	presenceData.state = "";
+	presenceData.smallImageText = "Zgubił się...";
+	presenceData.smallImageKey = Assets.Question;
+}
+
+const operators = [
+	"pkp-intercity",
+	"polregio",
+	"arriva",
+	"leo-express",
+	"koleje-wielkopolskie",
+	"koleje-dolnoslaskie",
+	"koleje-mazowieckie",
+	"koleje-malopolskie",
+	"wkd",
+	"lka",
+	"koleje-slaskie",
+	"skm-trojmiasto",
+];
+
+presence.on("UpdateData", async () => {
+	const { href } = document.location,
+		presenceData: PresenceData = {
+			details: "Ładowanie.",
+			largeImageKey: Assets.Logo,
+			smallImageText: "Ładowanie...",
+			smallImageKey: Assets.Train,
+			startTimestamp: browsingTimestamp,
+		},
+		[privacySetting] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+		]);
+
+	let currentPath;
+
```

Comment:
Suggestion 4:
Let's start by organising everything under their hostname. After we can start an if else or a switch statement to check for things

```suggestion
switch (hostname) {
	case "koleo.pl": {
		// Code for koleo.pl
		break;
	}
	case "pomoc.koleo.pl": {
		// Code for pomoc.koleo.pl
		break;
	}
	case "skelp.koleo.pl": {
		// Code for sklep.koleo.pl
		break;
	}
	case "magazyn.koleo.pl": {
		// Code for magazyn.koleo.pl
		break;
	}
	case "travel.koleo.pl": {
		// Code for travel.koleo.pl
		break;
	}

	case "aplikacja.koleo.pl": {
		// Code for aplikacja.koleo.pl
		switch (true) {
			case pathname === "": {
				const startStation = document
					.querySelector("#query_start_station")
					?.textContent?.toLowerCase(),
					endStation = document.querySelector("#query_end_station")?.textContent?.toLowerCase();
				if (startStation && endStation) {
					presenceData.details = privacySetting ? "Viewing a route" : "Viewing route"
					presenceData.state = `From ${startStation} - To ${endStation}`
				} else presenceData.details = "Viewing the homepage"

				break;
			}
		}
		break;
	}
}
```